src/api/graphql.ts

```diff
@@ -103,22 +109,82 @@ export async function getDeployedContractsByWallet(
   };
 }
 
-export async function getWalletInteractionsForContract(
-  arweave: Arweave,
-  params: {
-    address?: string | undefined;
-    contractTxId: string;
-    blockHeight: number | undefined;
+class ContractInteractionsCacheKey {
+  constructor(
+    public readonly arweave: Arweave,
+    public readonly contractTxId: string,
+    public readonly blockHeight: number | undefined,
+    public readonly address?: string | undefined,
+    public readonly logger?: winston.Logger | undefined,
+  ) {}
+
+  toString(): string {
+    return `${this.contractTxId}-${
+      this.blockHeight ? `this.blockHeight` : 'null'
+    }${this.address ? `-${this.address}` : ''}`;
+  }
+
+  // Facilitate ReadThroughPromiseCache key derivation
+  toJSON() {
+    return { cacheKey: this.toString() };
+  }
+}
+
+// Cache contract interactions for 60 seconds since block time is around 2 minutes
+const contractInteractionsCache: ReadThroughPromiseCache<
+  ContractInteractionsCacheKey,
+  { interactions: Map<TransactionID, GenericContractInteraction> }
+> = new ReadThroughPromiseCache({
+  cacheParams: {
+    cacheCapacity: 10_000,
+    cacheTTL: 1000 * 60, // 60 seconds
   },
+  readThroughFunction: readThroughToWalletInteractionsForContract,
+});
+
+export async function getWalletInteractionsForContract({
+  arweave,
+  contractTxId,
+  address,
+  blockHeight,
+  logger,
+}: {
+  arweave: Arweave;
+  contractTxId: string;
+  address?: string | undefined;
+  blockHeight?: number | undefined;
+  logger?: winston.Logger | undefined;
```

Comment:
nit: can be optional `?` but cannot be `undefined`
```suggestion
  logger?: winston.Logger;
```